PowerEditor/src/Utf8_16.cpp

```diff
@@ -442,6 +457,15 @@ void Utf8_16_Write::fclose()
 
 	if (m_pFile)
 	{
+		if (!keepOriginalLength && !m_bWriteFailure)

```

Comment:
> _chsize_s is called ONLY when it needs to be called (when the file is shortened/shrunk).

No, it's not what I want. As what I said, _chsize_s should be called only after writing on  disk failure.
This condition let all normal cases have this special treatment.

Reply:
@donho ,

Calling `_chsize_s ` after the write failure is pointless (or very bad hack to say the least if your idea is to extend the file to its initial length after the fail).